packages/angular-sdk-components/src/lib/_components/field/user-reference/user-reference.component.ts

```diff
@@ -106,22 +114,27 @@ export class UserReferenceComponent implements OnInit, OnDestroy {
   }
 
   // Callback passed when subscribing to store change
-  onStateChange() {
-    this.checkAndUpdate();
+  async onStateChange() {
+    await this.checkAndUpdate();
+  }
+
+  private _filter(value: string): string[] {
+    const filterVal = (value || this.filterValue).toLowerCase();
+    return this.options$?.filter(option => option.value?.toLowerCase().includes(filterVal));
   }
 
-  checkAndUpdate() {
+  async checkAndUpdate() {
     // Should always check the bridge to see if the component should
     // update itself (re-render)
     const bUpdateSelf = this.angularPConnect.shouldComponentUpdate(this);
 
     // ONLY call updateSelf when the component should update
     if (bUpdateSelf) {
-      this.updateSelf();
+      await this.updateSelf();
     }
   }
 
-  updateSelf() {
+  async updateSelf() {
```

Comment:
Instead of adding async for all the methods move the api call to seperate method and add it there.

Reply:
I had tried that but in this case it is not working, data needs to be available when _filter is triggered by observable. unless i make wait on checkAndUpdate this is not working. This code is similar to autocomplete code.